README.md

```diff
@@ -1,31 +1,403 @@
+|**`ScalprumProvider`** module|The **`ScalprumProvider`** is the main root component. If you want to load modules, you must ensure they are enclosed in the `<ScalprumProvider />` tags. `ScalprumProvider` has a mandatory `config` field. This informs `ScalprumProvider` of which remote modules are available to render. For the purpose of the demo, `const config: AppsConfig = {}`, we set up `remoteModule` to use in the next steps. The config requires a `name`, and a `manifestLocation`.|
+|**`@scalprum/react-core`**|The **`@scalprum/react-core`** packages provide react core bindings for `@scalprum/core`.|
+|**`ScalprumComponent`**|The **`ScalprumComponent`** is provided by the **`@scalprum/react-core`** packages. It requires two parameters: **`scope`**: the `name` that you set in the `ScalprumProvider`'s `config{}`. **`module`**: the actual module that you want to render. For the purposes of this demo, create `RemoteModuleComponent`.
+|
+
+
+2. Edit the `index.tsx` file and make the following changes: 
+
+```jsx
+import React from 'react';
+import { createRoot } from 'react-dom/client';
+import ScalprumRoot from './ScalprumRoot';
+
+const App = () => {
+  return (
+    <ScalprumRoot/>
+  )
+}
+
+const domNode = document.getElementById('root');
+if(domNode) {
+  const root = createRoot(domNode);
+  root.render(<App />)
+}
+```
+
+3. In the `webpack.config.js` file, add the `moduleFederation` plugin to set up the environment for plugin applications to use the context of the host applications: 
+
+```jsx
+// declare shared dependencies
+const moduleFederationPlugin = new ModuleFederationPlugin({
+  name: 'host',
+  filename: 'host.[fullhash].js',
+  shared: [
+    {'@openshift/dynamic-plugin-sdk': { singleton: true, requiredVersion: '*'} },
+    { '@scalprum/react-core': { singleton: true, requiredVersion: '*'} },
+    { react: { singleton: true, requiredVersion: '*' } },
+    { 'react-dom': { singleton: true, requiredVersion: '*' } },
+    // any other packages you wish to share
+  ]
+})
+```
+
+Note that for demo purposes, the `requiredVersion` for dependencies in this example is set to `*`, but you can update this to a version that suits your requirements.
+
+4. In the `webpack.config.js` file, add an entry also for the `moduleFederationPlugin` to the declared plugins: 
+
+```jsx
+   plugins: [
+    new HtmlWebpackPlugin({
+      template: "index.html",
+    }),
+    new MiniCssExtractPlugin(),
+    // scalprum required plugin
+    moduleFederationPlugin,
+  ],
+```
+5. To create the remote module, in the `scalprum-demo` directory, create a **remoteModule** directory. 
+6. Install the following dependencies: 
+```shell
+$ npm i --save-dev @openshift/dynamic-plugin-sdk-webpack
+```
+7. In the **scalprum-demo/remoteModule** directory, create a `webpack.config.js` file. 
+8. You can copy the contents of `webpack.config.js` file from the previous steps.
+9. In the `webpack.config.js` file, remove the `HtmlWebpackPlugin` because it breaks dynamic modules.
```

Comment:
Just a note, we are working on fixing the issues with the `HtmlWebpackPlugin`.

Reply:
Do we have a public issue. I can add "Until x issue is resolved".

Reply:
Not public I am afraid.